### PR TITLE
[TASK] Add shell methods, bash & zsh, install some software

### PR DESCRIPTION
**Pull request by Cyb10101**
SSH Key should be better. I'll do it another time.

### CODE REVIEW

File: .zshrc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ZSH_THEME="cyb"
+
+# Uncomment the following line to use case-sensitive completion.
+# CASE_SENSITIVE="true"
+
+# Uncomment the following line to use hyphen-insensitive completion.
+# Case sensitive completion must be off. _ and - will be interchangeable.
+# HYPHEN_INSENSITIVE="true"
+
+DISABLE_AUTO_UPDATE="false"
```
Comment (Kanti) [inline .zshrc]: .

File: .zshrc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+export UPDATE_ZSH_DAYS=90
+
+# Which plugins would you like to load? (plugins can be found in ~/.oh-my-zsh/plugins/*)
+# Custom plugins may be added to ~/.oh-my-zsh/custom/plugins/
+# Example format: plugins=(rails git textmate ruby lighthouse)
+# Add wisely, as too many plugins slow down shell startup.
+plugins=(git)
```
Comment (Kanti) [inline .zshrc]: .

File: Dockerfile [hunk 3]
```diff
@@ -1,28 +1,50 @@
...
+    echo "deb http://deb.debian.org/debian stretch universe" >> /etc/apt/sources.list && \
+    echo "deb http://deb.debian.org/debian stretch-updates universe" >> /etc/apt/sources.list && \
+    echo "deb http://deb.debian.org/debian stretch multiverse" >> /etc/apt/sources.list && \
+    echo "deb http://deb.debian.org/debian stretch-updates multiverse" >> /etc/apt/sources.list && \
+    apt-get update && apt-get -y dist-upgrade && \
+    apt-get install -y sudo less vim nano diffutils tree git-core bash-completion zsh htop && \
+    rm -rf /var/lib/apt/lists/* && \
+    usermod -aG sudo application && \
+    echo "%sudo ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers && \
+    update-alternatives --set editor /usr/bin/vim.basic
```
Comment (Kanti) [inline Dockerfile]: 👍

File: Dockerfile [hunk 4]
```diff
@@ -1,28 +1,50 @@
 FROM webdevops/php-apache-dev:7.3
 
-# add sudo; without password
-RUN apt-get update && \
-  apt-get install -y sudo vim nano tree bash-completion && \
-  rm -rf /var/lib/apt/lists/* && \
-  usermod -aG sudo application && \
-  echo "%sudo ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
+RUN \
+    echo "deb http://deb.debian.org/debian stretch universe" >> /etc/apt/sources.list && \
```
Comment (Kanti) [inline Dockerfile]: 👎

File: Dockerfile [hunk 5]
```diff
@@ -1,28 +1,50 @@
...
+    echo "deb http://deb.debian.org/debian stretch-updates universe" >> /etc/apt/sources.list && \
+    echo "deb http://deb.debian.org/debian stretch multiverse" >> /etc/apt/sources.list && \
+    echo "deb http://deb.debian.org/debian stretch-updates multiverse" >> /etc/apt/sources.list && \
+    apt-get update && apt-get -y dist-upgrade && \
```
Comment (Kanti) [inline Dockerfile]: 👎

File: Dockerfile [hunk 6]
```diff
@@ -1,28 +1,50 @@
...
+
+# Configure user
 USER application
 RUN composer global require hirak/prestissimo
 
-# add .git-completion.bash
-RUN curl https://raw.githubusercontent.com/git/git/v$(git --version | awk 'NF>1{print $NF}')/contrib/completion/git-completion.bash > /home/application/.git-completion.bash
-# add .git-prompt.bash
-RUN curl https://raw.githubusercontent.com/git/git/v$(git --version | awk 'NF>1{print $NF}')/contrib/completion/git-prompt.sh > /home/application/.git-prompt.sh
+RUN echo "source ~/.shell-methods" >> ~/.bashrc && \
```
Comment (Kanti) [inline Dockerfile]: move into .additional_bashrc.sh

File: Dockerfile [hunk 7]
```diff
@@ -1,28 +1,50 @@
...
+    echo "sshAgentRestart" >> ~/.bashrc && \
+    echo "sshAgentAddKey 7d ~/.ssh/id_rsa" >> ~/.bashrc && \
+    echo "stylePS1" >> ~/.bashrc && \
+    echo "bashCompletion" >> ~/.bashrc && \
+    git clone https://github.com/robbyrussell/oh-my-zsh.git ~/.oh-my-zsh
 
-# add .additional_bashrc.sh
-COPY .additional_bashrc.sh /home/application/.additional_bashrc.sh
+COPY .shell-methods .zshrc .oh-my-zsh/custom/themes/cyb.zsh-theme /home/application/
```
Comment (Kanti) [inline Dockerfile]: shorten path: ``.oh-my-zsh/custom/themes/cyb.zsh-theme``

File: .bashrc [hunk 8]
```diff
@@ -0,0 +1,17 @@
+source ~/.shell-methods
+sshAgentRestart
+sshAgentAddKey 7d ~/.ssh/id_rsa
+stylePS1
+bashCompletion
+
+export PATH=${PATH}:./bin:./vendor/bin:./node_modules/.bin
+
+# Maybe only for root?
+#if [[ $EUID -eq 0 ]]; then
```
Comment (Cyb10101) [inline .bashrc]: @Kanti Please check node, npm, yarn ... maybe docker-compose path missing

```
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

File: .shell-methods [hunk 9]
```diff
@@ -88,3 +88,12 @@ function addAlias {
     # less defaults
     alias less='less -FSRX'
 }
+
+function addDockerAlias {
```
Comment (Kanti) [inline .shell-methods]: @Cyb10101 is this function executable from within Bash/Zsh?

File: Dockerfile (hunk 5 above)
Comment (Cyb10101) [reply Dockerfile]: +1

File: Dockerfile (hunk 7 above)
Comment (Cyb10101) [reply Dockerfile]: +1

File: Dockerfile (hunk 6 above)
Comment (Cyb10101) [reply Dockerfile]: +1

File: Dockerfile (hunk 4 above)
Comment (Cyb10101) [reply Dockerfile]: +1

File: .zshrc (hunk 2 above)
Comment (Cyb10101) [reply .zshrc]: +1

File: .zshrc (hunk 1 above)
Comment (Cyb10101) [reply .zshrc]: +1

File: .bashrc-additional [hunk 10]
```diff
@@ -0,0 +1,16 @@
+source ~/.shell-methods
```
[the remark this answers is not in the document]
Comment (Cyb10101) [reply .bashrc-additional]: +1

File: .bashrc-additional [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+sshAgentRestart
+sshAgentAddKey 7d ~/.ssh/id_rsa
+stylePS1
+bashCompletion
+
+export PATH=${PATH}:./bin:./vendor/bin:./node_modules/.bin
+
+# Maybe only for root?
```
[the remark this answers is not in the document]
Comment (Cyb10101) [reply .bashrc-additional]: +1

File: .bashrc-additional [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+#if [[ $EUID -eq 0 ]]; then
```
[the remark this answers is not in the document]
Comment (Cyb10101) [reply .bashrc-additional]: +1

File: .bashrc-additional [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+    DOCKER_COMPOSE_PROJECT=$(sudo docker inspect ${HOSTNAME} | grep '"com.docker.compose.project":' | awk '{print $2}' | tr --delete '"' | tr --delete ',')
+    NODE_CONTAINER=$(sudo docker ps -f "name=${DOCKER_COMPOSE_PROJECT}_node_1" --format {{.Names}})
+    alias node='sudo docker exec -w $(pwd) -it ${NODE_CONTAINER} node'
+    alias npm='sudo docker exec -w $(pwd) -it ${NODE_CONTAINER} npm'
+    alias yarn='sudo docker exec -w $(pwd) -it ${NODE_CONTAINER} yarn'
+#fi;
```
[the remark this answers is not in the document]
Comment (Cyb10101) [reply .bashrc-additional]: +1

File: .shell-methods (hunk 9 above)
Comment (Cyb10101) [reply .shell-methods]: yes

**Comment by ChrisB9** [comment]
this pull request will be closed for now as i cannot accept it like that.
We need this image to be basic to be adaptable by many for numerous projects therefore everything objective will not be added to it. That includes zsh (even though i personally like zsh, it does not belong into a base image). Therefore please split this request into smaller ones that provide a more general use-case and does not bloat this image

**Pull request closed without merging**